### PR TITLE
Migrating Keyword False Positives

### PR DESCRIPTION
**Pull request by domanchi**
`KeywordDetector` had a variety of different false positive heuristics that only applied to `KeywordDetector`. This PR moves it to the new filters architecture.

It also drops support for a hard-coded list of false positives (e.g. `fake`) given that we have `detect_secrets.filters.wordlist`.

### CODE REVIEW

File: tests/plugins/keyword_test.py [hunk 1]
```diff
@@ -227,105 +218,24 @@ def test_analyze_objective_c_positives(self, file_content):
...
-                '.yaml',
-                '.yml',
-            )
-        ),
-    )
-    def test_analyze_yaml_negatives(self, file_content, file_extension):
-        assert not KeywordDetector().analyze_line(
-            filename=f'mock_filename{file_extension}',
-            line=file_content,
-        )
```
Comment (domanchi) [inline tests/plugins/keyword_test.py:316]: @KevinHock : do you remember why this test was supposed to behave as such? I spent some time reading the code, but I couldn't understand why YAML files would be treated differently, and that lines that would have been caught in a non-YAML file will be ignored with YAML files.

Comment (KevinHock) [reply tests/plugins/keyword_test.py:316]: I added {{ to prefix the standard positives test, and think I made this test to test we skipped them.

I think this had to do specifically with 
```
        ) or (
            filetype == FileType.YAML
            and lowered_secret.startswith('{{')
            and lowered_secret.endswith('}}')
        )
```
at the time

https://github.com/Yelp/detect-secrets/commit/592d077084f4f66d4175d2e6b7055a8529f0ade7#diff-e6de82b29d51ff0d50bb9a4951479046f43f6dad3d40ba27e3dd7e2a9f6fda8bR226

Comment (domanchi) [reply tests/plugins/keyword_test.py:316]: Yeah, that's so bizarre.

Looks like it was [removed](https://github.com/Yelp/detect-secrets/commit/a53d12e65c450f5daea44290e7ba0cee744fe686#diff-e6de82b29d51ff0d50bb9a4951479046f43f6dad3d40ba27e3dd7e2a9f6fda8bL308-L310) though, so I don't know how tests kept passing.

I'm going to leave it out for now, and migrate the filters based on the master branch, and the active false positive filters on this plugin.

File: detect_secrets/filters/heuristic.py [hunk 2]
```diff
@@ -113,3 +113,53 @@ def is_non_text_file(filename: str) -> bool:
         '.zip',
     ),
 )
+
+
+def is_templated_secret(secret: str) -> bool:
+    """
+    Filters secrets that are shaped like: {secret}, <secret>, or ${secret}.
+    """
+    if secret[0] == '$':
```
Comment (tanx16) [inline detect_secrets/filters/heuristic.py:122]: Did you forget to remove this after copying it to the new `is_prefixed_with_dollar_sign ` function?

File: tests/filters/heuristic_filter_test.py [hunk 3]
```diff
@@ -79,3 +81,49 @@ def test_success(self, secret, line):
...
         assert not filters.heuristic.is_likely_id_string(secret, line)
+
+
+@pytest.mark.parametrize(
+    'line, result',
+    (
+        ('secret = $hunter2', False),
+        ('secret = {hunter2}', False),
+        ('secret = <hunter2>', False),
+        ('secret = hunter2', True),
```
Comment (tanx16) [inline tests/filters/heuristic_filter_test.py:92]: Maybe add a test for `secret= ${hunter2}`?

File: detect_secrets/filters/heuristic.py [hunk 4]
```diff
@@ -113,3 +113,53 @@ def is_non_text_file(filename: str) -> bool:
...
+        return True
+
+    if (
+        (secret[0] == '{' and secret[-1] == '}')
+        or (secret[0] == '<' and secret[-1] == '>')
+        or (secret[0] == '$' and secret[1] == '{' and secret[-1] == '}')
```
Comment (tanx16) [inline detect_secrets/filters/heuristic.py:128]: Is it possible for a secret to be 1 character and `IndexError`ing on this line?

File: detect_secrets/filters/heuristic.py (hunk 2 above)
Comment (domanchi) [reply detect_secrets/filters/heuristic.py:122]: You're absolutely right. ++

File: detect_secrets/filters/heuristic.py (hunk 4 above)
Comment (domanchi) [reply detect_secrets/filters/heuristic.py:128]: It's a fair point, but like super unlikely. Entropy scanners definitely require more than one character to trip it up, and regexes should require more than one character to meet the bar. After all, a one-character secret is a terrible secret.

I'll add a try/except clause for any keyword filter trip ups.

**Review by tanx16 — APPROVED** [review]

**Pull request merged**